src/vm/i386/asmhelpers.S

```diff
@@ -472,10 +472,12 @@ NESTED_END CallDescrWorkerInternal, _TEXT
 #ifdef _DEBUG
 // int __fastcall HelperMethodFrameRestoreState(HelperMethodFrame*, struct MachState *)
 LEAF_ENTRY HelperMethodFrameRestoreState, _TEXT
+    #define     RETN 8
```

Comment:
Could you please disass the whole function that called the LazyMachStateCaptureState in your example where there was the sub  $0x4,%esp?